vendor/k8s.io/kube-openapi/pkg/handler/handler.go

```diff
@@ -202,3 +211,53 @@ func toGzip(data []byte) []byte {
 	zw.Close()
 	return buf.Bytes()
 }
+
+// RegisterOpenAPIVersionedService registers a handler to provides access to provided swagger spec.
+func RegisterOpenAPIVersionedService(openapiSpec *spec.Swagger, servePath string, handler common.PathHandler) (*OpenAPIService, error) {
+	o := OpenAPIService{}
+	if err := o.UpdateSpec(openapiSpec); err != nil {
+		return nil, err
+	}
+
+	files := map[string]func() ([]byte, string, time.Time){
+		"application/json":                                           o.getSwaggerBytes,
+		"application/com.github.proto-openapi.spec.v2@v1.0+protobuf": o.getSwaggerPbBytes,
+	}
+
+	handler.Handle(servePath, gziphandler.GzipHandler(http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			var getDataAndETag func() ([]byte, string, time.Time)
+			// Get the first value associated with header "Accept"
+			decipherableFormat := r.Header.Get("Accept")
+			if decipherableFormat == "" || decipherableFormat == "*/*" {
+				// If "Accept" header is not set or "*/*", serve json file as default
+				decipherableFormat = "application/json"
+			}
+			w.Header().Add("Vary", "Accept")
+
```

Comment:
This needs to work with actual Accept headers, which are allowed to specify a prioritized list of accepted content types. For example, a request could indicate it accepts some unknown protobuf format, the known protobuf format, yaml, and json, and the server would be expected to return the known protobuf format

Reply:
For example, see parsing/prioritization/matching in https://github.com/kubernetes/kubernetes/blob/228b7d5d53c404f8d6f8b2031fcf6abb30a0684e/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go#L284

Reply:
Thanks! PR sent in kube-openapi repo to use that kubernetes/kube-openapi#36